core/src/main/java/com/sequenceiq/cloudbreak/controller/ClusterCreationSetupService.java

```diff
@@ -141,25 +155,34 @@ public Cluster prepare(ClusterRequest request, Stack stack, Blueprint blueprint,
         return savedCluster;
     }
 
-    private List<ClusterComponent> addAmbariRepoConfig(Optional<Component> stackAmbariRepoConfig, List<ClusterComponent> components,
+    private void checkVDFFile(ClusterComponent ambariRepoConfig, ClusterComponent hdpRepoConfig) throws IOException {
+        AmbariRepo ambariRepo = ambariRepoConfig.getAttributes().get(AmbariRepo.class);
+
+        if (ambariRepositoryVersionService.isNewerOrEqualAmbariApi(AMBARI_VERSION_2_6_0_0, ambariRepo::getVersion)
+                && !containsVDFUrl(hdpRepoConfig.getAttributes())) {
+            throw new BadRequestException("Couldn't determine any VDF file");
+        }
+
+    }
+
+    private ClusterComponent addAmbariRepoConfig(Optional<Component> stackAmbariRepoConfig,
```

Comment:
I would rename it to createAmbariRepoConfig or determineAmbariRepoConfig, this method doesn't "add" anything anymore.
Same for the "addHDPRepoConfig" and "addAmbariDatabaseConfig" method.